### PR TITLE
Docs about generations.

### PR DESCRIPTION
**Pull request by teh**

### CODE REVIEW

File: basics/undo-installation.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Undo installation
+
+Nix supports atomic *rollback* of your packages to a previous state. *Rollback* is the same as undoing your last action (or actions).
+
```
Comment (Keats) [inline basics/undo-installation.md:4]: Maybe a quick explanation of what is a generation in the paragraph above. I guess it's like the git reflog for environment change? Could use a glossary definition as well

File: basics/undo-installation.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ 3     2015-09-24 10:49:51
+```
+
+With generation 3 still around We can switch back to it like so:
+
+```console
+$ nix-env --switch-generation 3
+```
+
+With `--switch-generation` we can in fact switch to any generation, e.g. your install fro 4 weeks ago.
```
Comment (Keats) [inline basics/undo-installation.md]: from

File: basics/undo-installation.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+```
+
+Let's undo our last change (e.g. the installation of `hello`):
+
+```console
+$ nix-env --rollback
+switching from generation 3 to 2
+```
+
+Checking with `--list-generations` we can see that we're now the previous install (2). Note that generation 3 is still around.
```
Comment (Gautier) [inline basics/undo-installation.md]: I suspect a word is missing from the first sentence.

File: basics/undo-installation.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+```console
+$ nix-env --list-generations
+ 1     2015-09-21 23:05:34
+ 2     2015-09-22 17:10:56   (current)
+ 3     2015-09-24 10:49:51
+```
+
+With generation 3 still around We can switch back to it like so:
```
Comment (Gautier) [inline basics/undo-installation.md]: "With generation 3 still around, we can switch back to it like so:"

File: basics/undo-installation.md (hunk 1 above)
Comment (teh) [inline basics/undo-installation.md:4]: done

File: basics/undo-installation.md (hunk 2 above)
Comment (teh) [inline basics/undo-installation.md]: done

File: basics/undo-installation.md (hunk 3 above)
Comment (teh) [inline basics/undo-installation.md]: done

File: basics/undo-installation.md (hunk 4 above)
Comment (teh) [inline basics/undo-installation.md]: done

**Comment by teh** [comment]
PTAL

File: GLOSSARY.md [hunk 5]
```diff
@@ -3,3 +3,7 @@
 The Nix package database is a set of nested records, with the top-level record usually being `nixpkgs`.
 Accessing attributes in nested records is done by chaining names with a dot, just like in many other languages (e.g. `nixpkgs.pythonPackages.django`).
 This full "path" to a package is called the "attribute path".
+
+# generation
+
+Every time Nix modifies the system, e.g. by installing or removing a package, it creates a new generation. One can go back to a previous generation of the sytem with the command `nix-env --switch-generation`.
```
Comment (Keats) [inline GLOSSARY.md]: maybe add X at the end of the command to indicate it needs an arg?

Comment (teh) [inline GLOSSARY.md]: done

**Pull request merged**